osu.Game/Overlays/Toolbar.cs

```diff
@@ -39,13 +47,39 @@ protected override void PopOut()
             FadeOut(transition_time, EasingTypes.InQuint);
         }
 
-        public override void Load(BaseGame game)
+        private void CheckUser()
         {
-            base.Load(game);
+            if (user == null)
+                return;
+            if (user.UserId == 0)
+            {
+                toolbarUserButton.UpdateButton(1, "Not signed in");
+            }
+            else if (user.UserId != toolbarUserButton.UserId)
+            {
+                toolbarUserButton.UpdateButton(user.UserId, user.Name);
+                toolbarUserButton.UserId = user.UserId;
+            }
+        }
 
-            RelativeSizeAxes = Axes.X;
-            Size = new Vector2(1, height);
+        private void InitUser()

```

Comment:
you're calling this every 10 seconds with no checks for whether it's still running, failure of last call, etc.

what?
what?
what?